src/core/mesh/qgsmesh3daveraging.cpp

```diff
@@ -151,3 +248,513 @@ int QgsMeshSingleLevelAveragingMethod::verticalLevel() const
 {
   return mVerticalLevel;
 }
+
+bool QgsMeshSingleLevelAveragingMethod::hasValidInputs() const
+{
+  return mVerticalLevel >= 1;
+}
+
+void QgsMeshSingleLevelAveragingMethod::volumeRangeForFace(
+  double &startVerticalLevel,
+  double &endVerticalLevel,
+  const QVector<double> verticalLevels ) const
+{
+  if ( ( mVerticalLevel - 1 >= 0 ) && ( mVerticalLevel < verticalLevels.size() ) )
```

Comment:
( mVerticalLevel < verticalLevels.size() )  is this correct? shouldn't it be <= if the level is 1..N ?

Reply:
verticalLevels has size +1 (vertical levels are for all 3d faces so it has size nVolumes + 1 for each face)